client/lib/api/user_preferences.dart

```diff
@@ -92,8 +95,21 @@ class UserPreferences {
   }
 
   Future<bool> setNotificationsEnabled(bool value) async {
-    return (await SharedPreferences.getInstance())
+    var result = (await SharedPreferences.getInstance())
         .setBool(UserPreferenceKey.NotificationsEnabled.toString(), value);
+    await _updateFirebaseAutoInit();
+    return result;
+  }
+
+  Future<void> _updateFirebaseAutoInit() async {
+    // Updates the firebase auto init option. This method is called
+    // after a user modifies either analytics or notifications.
+    var firebaseMessaging = FirebaseMessaging();
+
+    // Enable Firebase Auto Init iff either notifications or analytics is enabled
+    var firebase_auto_init =
+        (await getNotificationsEnabled()) || (await getAnalyticsEnabled());
+    await firebaseMessaging.setAutoInitEnabled(firebase_auto_init);
```

Comment:
Remove the variable above and inline in. I think the indirection detracts from readability rather than adds to it.

```suggestion
    await FirebaseMessaging().setAutoInitEnabled(firebase_auto_init);
```